runtime/sunvmi/sunvmi.c

```diff
@@ -716,7 +716,16 @@ JVM_GetSystemPackages_Impl(JNIEnv* env)
 
 						funcs->internalEnterVMFromJNI(vmThread);
 						packageName = getPackageName(packageIDList[i], &packageNameLength);
-						packageString = funcs->catUtfToString4(vmThread, packageName, packageNameLength, (U_8*)"/", 1, NULL, 0, NULL, 0);
+						/*
+						 * java.lang.Package.getSystemPackages() expects a trailing slash in Java 8
+						 * but no trailing slash in Java 9.
+						 */
+						if (J2SE_VERSION(vm) >= J2SE_19) {
+							packageString = vmThread->javaVM->memoryManagerFunctions->j9gc_createJavaLangString(vmThread,
```

Comment:
Might as well use vm instead of vmThread->javaVM since vm is available.

Reply:
Done.